src/main/java/com/powsybl/openloadflow/network/AbstractLfBranch.java

```diff
@@ -39,6 +66,17 @@ protected AbstractLfBranch(LfBus bus1, LfBus bus2, PiModel piModel) {
         this.piModel = Objects.requireNonNull(piModel);
     }
 
+    protected static SortedSet<LfTemporaryLimit> createSortedTemporaryLimitsSet(CurrentLimits currentLimits, LfBus bus) {
+        SortedSet<LfTemporaryLimit> temporaryLimits = new TreeSet<>(TEMPORARY_LIMITS_COMPARATOR);
+        if (currentLimits != null) {
+            for (CurrentLimits.TemporaryLimit temporaryLimit : currentLimits.getTemporaryLimits()) {
+                double valuePerUnit = temporaryLimit.getValue() != Double.MAX_VALUE ? temporaryLimit.getValue() * bus.getNominalV() / PerUnit.SB : Double.MAX_VALUE;
```

Comment:
`Double.NaN`?

Reply:
Indeed I think that we don't need to make the distinction between temporary and permanent limit. The permanent limit is just a temporary limit with infinite duration time. And with that, the comparator works for all limits and it will ease the post processing on limit violations.